Sources/LiveViewNative/Modifiers/Text Input and Output Modifiers/TextContentTypeModifier.swift

```diff
@@ -0,0 +1,300 @@
+
+    func body(content: Content) -> some View {
+        switch textContentType {
+        #if !os(macOS)
+        case .url:
+            content.textContentType(.URL)
+        case .name_prefix:
+            content.textContentType(.namePrefix)
+        case .name:
+            content.textContentType(.name)
+        case .name_suffix:
+            content.textContentType(.nameSuffix)
+        case .given_name:
+            content.textContentType(.givenName)
+        case .middle_name:
+            content.textContentType(.middleName)
+        case .family_name:
+            content.textContentType(.familyName)
+        case .nickname:
+            content.textContentType(.nickname)
+        case .organization_name:
+            content.textContentType(.organizationName)
+        case .job_title:
+            content.textContentType(.jobTitle)
+        case .location:
+            content.textContentType(.location)
+        case .full_street_address:
+            content.textContentType(.fullStreetAddress)
+        case .street_address_line_1:
+            content.textContentType(.streetAddressLine1)
+        case .street_address_line_2:
+            content.textContentType(.streetAddressLine2)
+        case .address_city:
+            content.textContentType(.addressCity)
+        case .address_city_and_state:
+            content.textContentType(.addressCityAndState)
+        case .address_state:
+            content.textContentType(.addressState)
+        case .postal_code:
+            content.textContentType(.postalCode)
+        case .sublocality:
+            content.textContentType(.sublocality)
+        case .country_name:
+            content.textContentType(.countryName)
+        case .new_password:
+            content.textContentType(.newPassword)
+        case .email_address:
+            content.textContentType(.emailAddress)
+        case .telephone_number:
+            content.textContentType(.telephoneNumber)
+        case .credit_card_number:
+            content.textContentType(.creditCardNumber)
+        #elseif os(iOS) || os(tvOS)
+        case .date_time:
+            content.textContentType(.dateTime)
+        case .flight_number:
+            content.textContentType(.flightNumber)
+        case .shipment_tracking_number:
+            content.textContentType(.shipmentTrackingNumber)
+        #endif
+        case .username:
+            content.textContentType(.username)
+        case .password:
+            content.textContentType(.password)
+        case .one_time_code:
+            content.textContentType(.oneTimeCode)
+        case .none:
+            content.textContentType(nil)
+        default:
+            content
+        }
+    }
+
+    enum CodingKeys: String, CodingKey {
+        case textContentType = "text_content_type"
+    }
+}
+
+/// A content type to use with ``TextContentTypeModifier``.
+#if swift(>=5.8)
+@_documentation(visibility: public)
+#endif
+private enum TextContentType: String, Decodable {
```

Comment:
Is there a particular reason for having our own TextContentType type, rather than using the platform one?

Reply:
If you look at the 3 modifiers in #328, there are separate platform types for iOS, Mac, and Watch. This unifies them into one Elixir modifier that translates them to the right form for the platform. I adapted the technique Carson has been using in examples like `TabViewStyleModifier`.

Reply:
Ah, gotcha